### PR TITLE
Port QA projects to use javaRestTest and yamlRestTest

### PR DESCRIPTION
**Pull request by breskeby**
Do not use deprecated elasticsearch.rest-test plugin

### CODE REVIEW

File: qa/ccs-unavailable-clusters/build.gradle [hunk 1]
```diff
@@ -5,16 +5,13 @@
...
  * Side Public License, v 1.
  */
-apply plugin: 'elasticsearch.internal-testclusters'
-apply plugin: 'elasticsearch.standalone-rest-test'
-apply plugin: 'elasticsearch.rest-test'
-apply plugin: 'elasticsearch.test-with-dependencies'
+apply plugin: 'elasticsearch.internal-java-rest-test'
 
-testClusters.matching { it.name == "integTest" }.configureEach {
+testClusters.matching { it.name == "javaRestTest" }.configureEach {
```
Comment (mark-vieira) [inline qa/ccs-unavailable-clusters/build.gradle:10]: Not sure why we used`matching` here when there's only ever one. Wouldn't `testClusters.named()` be more concise?

File: qa/system-indices/build.gradle [hunk 2]
```diff
@@ -23,3 +23,5 @@ testClusters.configureEach {
   setting 'xpack.security.autoconfiguration.enabled', 'false'
   user username: 'rest_user', password: 'rest-user-password'
 }
+
+tasks.named('test').configure { enabled = false}
```
Comment (mark-vieira) [inline qa/system-indices/build.gradle:27]: Why is this necessary?

**Review by mark-vieira — APPROVED** [review]
LGTM

File: qa/ccs-unavailable-clusters/build.gradle (hunk 1 above)
Comment (breskeby) [reply qa/ccs-unavailable-clusters/build.gradle:10]: this is because we create those cluster lazy as part of the task creation. That means that taskClusters.named would not find the cluster yet when hitting this config block initially

File: qa/system-indices/build.gradle (hunk 2 above)
Comment (breskeby) [reply qa/system-indices/build.gradle:27]: there were no tests we did this in other places. But you're right we should just rely on gradle skipping those tasks anyhow.

Comment (breskeby) [reply qa/system-indices/build.gradle:27]: removed
